src/NUnitFramework/framework/Internal/Commands/ConstructFixturePerTestCaseCommand.cs

```diff
@@ -0,0 +1,67 @@
+// ***********************************************************************
+// Copyright (c) 2019 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using NUnit.Framework.Interfaces;
+
+namespace NUnit.Framework.Internal.Commands
+{
+    /// <summary>
+    /// ConstructFixtureCommand constructs the user test object if necessary.
+    /// </summary>
+    public class ConstructFixturePerTestCaseCommand : BeforeTestCommand
+    {
+        /// <summary>
+        /// Constructs a fixture per test case
+        /// </summary>
+        /// <param name="innerCommand">The inner command to which the command applies</param>
+        public ConstructFixturePerTestCaseCommand(TestCommand innerCommand)
+            : base(innerCommand)
+        {
+            TestSuite testSuite = null;
+            TestFixture testFixture = null;
+
+            ITest currentTest = Test;
+            while (currentTest != null && (testSuite == null || testFixture == null))
+            {
+                testSuite = testSuite ?? currentTest as TestSuite;
+                testFixture = testFixture ?? currentTest as TestFixture;
+                currentTest = currentTest.Parent;
+            }
+
+            Guard.ArgumentValid(testSuite != null, "ConstructFixturePerTestCaseCommand must reference a TestSuite", nameof(innerCommand));
+
+            BeforeTest = (context) =>
```

Comment:
@avilv Is there some reason that you didn't use a BeforeAndAfter command? That way, the cleanup of the fixture instance could be handled automatically without needing a separate command to do it.

Reply:
i thought about it but, but i figured since the convention right now is to have a construct and a dispose command i should try and stick to it.. i don't mind changing it though

Reply:
That's the architecture for commands that relate to classes, because classes have two different command sequences, one for before and one for after.

But for commands that relate to methods, we generally use beforeandafter commands when there are symmetric things to do because there is only one command sequence constructed. I think this is a bit confusing because we are constructing a class as part of the execution of a method.

Reply:
ah.. that makes sense, so say i add the dispose logic in AfterTest, now its doing two things - 
should it still be called `ConstructFixturePerTestCaseCommand` ? 
or perhaps `HandleFixturePerTestCaseCommand`

Reply:
Or just `FixturePerTestCaseCommand`?

Anyway, the advantage of Before + After is that the system keeps track for you. After doesn't run unless before ran. If you use two commands, you're supposed to do that yourself.

Reply:
done